### PR TITLE
[FLOSS H3] Integration: caching macOS workflow

### LINKED ISSUE #4168: Workflow: cache macOS dependencies
I realized the macOS install dependencies step is not being cached. Am I wrong with that assumption?
If not, I would suggest to add "uses: actions/cache@v2" to save some time in this step.

### PR DESCRIPTION
**Pull request by 4ydan**
Caching dependencies to increase  github action speed. Closes #4168

## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

File: .github/workflows/macOS.yml [hunk 1]
```diff
@@ -72,49 +72,35 @@ jobs:
...
+        echo 'EXISTING_CELLAR<<EOF' >> $GITHUB_ENV
+        ls -1 /usr/local/Cellar >> $GITHUB_ENV
+        echo 'EOF' >> $GITHUB_ENV
+
+    - name: Cache dependencies
+      id: dependency-cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          /.npm
```
Comment (lawli3t) [inline .github/workflows/macOS.yml:87]: Is this correct? Seems like `~/.npm` would be the correct path

**Comment by lawli3t** [comment]
Good addition, should speed up Mac OS X significantly. Although for some reason `npm` is now missing in the pipeline?

File: .github/workflows/macOS.yml (hunk 1 above)
Comment (4ydan) [reply .github/workflows/macOS.yml:87]: true, ty

**Comment by markus2330** [comment]
@4ydan can you check how much it improved the performance?

**Comment by 4ydan** [comment]
@markus2330 
Installing dependencies with Github actions takes at the moment around 10 minutes(https://github.com/ElektraInitiative/libelektra/runs/4537064004?check_suite_focus=true).
Reading cached dependencies takes around 2 minutes(https://github.com/ElektraInitiative/libelektra/runs/4330045360?check_suite_focus=true). 
The speed up would be ~8 minutes, meaning 20%, if the overall run is ~40 minutes.

**Comment by markus2330** [comment]
Good to hear! :sparkling_heart: Hope to see this ready to merge soon!

**Comment by 4ydan** [comment]
Sry, with my current knowledge there is no way I can fix this problem. Maybe I will come back to it one day.

**Pull request closed without merging**